src/net/mod.rs

```diff
@@ -107,3 +110,39 @@ where
         }
     }
 }
+
+/// Get the MQTT prefix of a device.
+///
+/// # Args
+/// * `app` - The name of the application that is executing.
+/// * `mac` - The ethernet MAC address of the device.
+///
+/// # Returns
+/// The MQTT prefix used for this device.
+pub fn get_device_prefix(
+    app: &str,
+    mac: smoltcp_nal::smoltcp::wire::EthernetAddress,
+) -> String<consts::U128> {
+    let mac_string = {
+        let mut mac_string: String<consts::U32> = String::new();
```

Comment:
Why not `U18` then? Or do it together with the app name in one single `write!()`?

Reply:
This uses U32 because it's the largest power of two that supports the 17 byte string (5 hyphens + 2 * 6 characters). I'm inclined to leave it as-is for now since I don't see an apparent benefit to dropping down the buffer size.

Reply:
Also, this is done as a two-stage write to make the format clear and easy to process. Writing it all in one go would result in some weird looking format strings that would be hard to interpret. Feel free to propose alternatives though.

Reply:
All good!